### DISCUSSION TITLE
Allow additional import for the SWC modularized imports

### DISCUSSION THREAD

**Discussion by yhkk1** [vercel/next.js#38554] [Ideas]
### Describe the feature you'd like to request

I want to add an additional css source file import when modularizing the import statement of a React component (similar to that of [babel-plugin-import](https://www.npmjs.com/package/babel-plugin-import)), but it seems there is no way of doing so.

Example import in source code:
```
import { Header, Footer } from 'my-library/components/App'
```

Expected-transformed imports:

```
import Header from 'my-library/components/App/Header'
import 'my-library/components/App/Header/style'
import Footer from 'my-library/components/App/Footer'
import 'my-library/components/App/Footer/style'
```

### Describe the solution you'd like

I feel like a viable solution would be to allow additional import alongside the transform:

```
modularizeImports: {
  'my-library/?(((\\w*)?/?)*)': {
    transform: 'my-library/{{ matches.[1] }}/{{member}}',
    additionalImports: [
      'my-library/{{ matches.[1] }}/{{member}}/style',
    ],
  },
}
```
This approach fits my use case, but it might not be a generic way of solving this kind of problem. 

### Describe alternatives you've considered

Can't think of any at this moment.

**Comment by otakustay** [comment]
To addition, some libraries may have different naming cases between exports and module paths, like:

```js
import {AutoComplete} from 'antd';
// is actually transformed to a dash-case path
import AutoComplete from 'antd/es/auto-complete';
// with an additional style import
import 'antd/es/auto-complete/style';
```

In this case, we may need some "naming case placeholders" like `{{ dash(matches.[1]) }}` to satisfy all situations.

**Comment by vnghia187** [comment]
yes we need a way to import styles from module with modularizeImports, otherwise switching from Babel using https://www.npmjs.com/package/babel-plugin-import to SWC is really painful.

**Comment by leerob** [comment]
https://nextjs.org/docs/advanced-features/compiler#modularize-imports
